pkg/rest/apisurface.go

```diff
@@ -430,3 +437,82 @@ func retrieveOriginatingIdentity(r *http.Request) (*osb.OriginatingIdentity, err
 	}
 	return nil, fmt.Errorf("unable to find originating identity")
 }
+
+// writeResponse will serialize 'object' to the HTTP ResponseWriter
+// using the 'code' as the HTTP status code
+func (s *APISurface) writeResponse(w http.ResponseWriter, code int, object interface{}) {
```

Comment:
I worry a little bit about these methods accumulating logic, but I can't see a better way at this point than making them methods with `APISurface` as a receiver.

Reply:
only way i can see is to always pass in the `enableCORS` as a boolean to the `writeResponse`. Or you'd have to get fancy with functions or middleware. So I think this is okay.